eox_tagging/api/v1/filters.py

```diff
@@ -26,22 +36,36 @@ class Meta:
         model = Tag
         fields = ["key", "status", "tag_type", "tag_value"]
 
+    @staticmethod
+    def _map_to_filter_params(search_field, value):
+        """
+        Used to parse search strings to fields/values related to the queryset to search.
+
+        Example usage:
+            >>> filter_params = self._parse_search_string("certificate_id", "1")
+            >>> {"target_id": {"id": 1}, "target_type": ""generatedcertificate"}
+        """
+        filter_params = {
+            "target_type": FILTER_TARGET_TYPES.get(search_field),
+        }
+        if search_field.startswith("certificate_"):
```

Comment:
this whole block still is super convoluted.
Basically you want `value` for every single case but `certificate_id`, why dont you make the if be `if name ='certificate_id` and be done with it?

Reply:
Because we need to search the queryset with `id` and not `certificate_id`

Reply:
Yes, I see that, but you are still making it convoluted for no reason.

For 4 fields you need:

```
"course_id": -> {"course_id": value}
"username": -> {"username": value}
"certificate_key": -> {"certificate_key": value}
"certificate_url": -> {"certificate_url": value}
```

but the last field requires
```
"certificate_id": -> {"id": int(value)}
```

This is why I ask: why not make it:

```
if search_field in ["certificate_id"]:
            filter_params["target_id"] = {
                "id": int(value)
            }
else:
            filter_params["target_id"] = {
                search_field: value,
            }

```

Which makes it clear about which fields require something special.
Right now,to understand the code I need to imagine the logic of this block on all the possible fields just so 1 of them can be singled out.

Reply:
Thanks for the explanation! Now I understand better what you're saying.

The thing is we need to do the _transformation_ with all the fields that start with `certificate`: certificate_download_url -> download_url, certificate_key -> key, certificate_id -> id. So the `target_id` can be:

1. `{"id": int(value)}`
2. `{"key": value}`
3. `{"download_url":value}`

I'll be checking again tomorrow to see if I miss something.

Reply:
The _transformation_ also includes:

"certificate_key": -> {"key": value}
"certificate_download_url": -> {"download_url": value}

I'll be checking this tomorrow to see if we can accomplish something better.